torchgeo/datamodules/vhr10.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""NWPU VHR-10 datamodule."""
+
+from typing import Any, Union
+
+import kornia.augmentation as K
+import torch
+
+from ..datasets import VHR10
+from ..samplers.utils import _to_tuple
+from ..transforms import AugmentationSequential
+from .geo import NonGeoDataModule
+from .utils import AugPipe, collate_fn_detection, dataset_split
+
+
+class VHR10DataModule(NonGeoDataModule):
+    """LightningDataModule implementation for the VHR10 dataset.
+
+    .. versionadded:: 0.6
+    """
+
+    std = torch.tensor(255)
+
+    def __init__(
+        self,
+        batch_size: int = 64,
+        patch_size: Union[tuple[int, int], int] = 512,
+        num_workers: int = 0,
+        val_split_pct: float = 0.2,
+        test_split_pct: float = 0.2,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize a new VHR10DataModule instance.
+
+        Args:
+            batch_size: Size of each mini-batch.
+            patch_size: Size of each patch, either ``size`` or ``(height, width)``.
+            num_workers: Number of workers for parallel data loading.
+            val_split_pct: Percentage of the dataset to use as a validation set.
+            test_split_pct: Percentage of the dataset to use as a test set.
+            **kwargs: Additional keyword arguments passed to
+                :class:`~torchgeo.datasets.VHR10`.
+        """
+        super().__init__(VHR10, batch_size, num_workers, **kwargs)
+
+        self.val_split_pct = val_split_pct
+        self.test_split_pct = test_split_pct
+        self.patch_size = _to_tuple(patch_size)
+
+        self.collate_fn = collate_fn_detection
+
+        self.train_aug = AugPipe(
+            AugmentationSequential(
+                K.Normalize(mean=self.mean, std=self.std),
+                K.Resize(self.patch_size),
+                K.RandomHorizontalFlip(),
+                K.ColorJiggle(0.1, 0.1, 0.1, 0.1, p=0.7),
+                K.RandomVerticalFlip(),
+                data_keys=["image", "boxes", "masks"],
+            ),
+            batch_size,
+        )
+        self.aug = AugPipe(
+            AugmentationSequential(
+                K.Normalize(mean=self.mean, std=self.std),
+                K.Resize(self.patch_size),
+                data_keys=["image", "boxes", "masks"],
+            ),
+            batch_size,
+        )
+
+    def setup(self, stage: str) -> None:
+        """Set up datasets.
+
+        Args:
+            stage: Either 'fit', 'validate', 'test', or 'predict'.
+        """
+        self.dataset = VHR10(**self.kwargs)
```

Comment:
So we never use the negative examples? Should we?

Reply:
Correct. We do not use them.

Whether we should is an interesting question. Practically, there isn't a way to pass 'no-object' images into detectors like Faster-RCNN (or any other that I know of). This follows an assumption that those detectors already account for negative samples by considering regions of the image where the object is absent.  Additionally negative regions (where object is absent) usually outnumber positive regions (object is present). 

So, if you were building a cat detector, you don't need to pass in images with no cats at all since parts of the image that don't contain a cat is enough for the network to distinguish between what is a cat and what is not. Whether those negative regions alone are enough for a detector deployed in the wild varies case by case.